testing/test_warnings.py

```diff
@@ -774,3 +775,53 @@ def test_it():
                 "*Unknown pytest.mark.unknown*",
             ]
         )
+
+
+def test_resource_warning(pytester: Pytester, monkeypatch: pytest.MonkeyPatch) -> None:
+    # Some platforms (notably PyPy) don't have tracemalloc.
+    # We choose to explicitly not skip this in case tracemalloc is not
+    # available, using `importorskip("tracemalloc")` for example,
+    # because we want to ensure the same code path does not break in those platforms.
+    try:
+        import tracemalloc  # noqa
+
+        has_tracemalloc = True
+    except ImportError:
+        has_tracemalloc = False
+
+    pytester.makepyfile(
```

Comment:
I think this test (tracemalloc not enabled) will fail if the person running the *pytest* test suite enables tracemalloc.

Reply:
Hmm why do you say that? And do you mean running in CPython?

I tried it here:

```
python -X tracemalloc=20 -m pytest testing\test_warnings.py
```

And it passes.

Reply:
Note the try/except is checking if tracemalloc is available at all, which is not on PyPy for example, not if tracemalloc is enabled or not.

Reply:
I guess `python -X tracemalloc=20` doesn't transfer, but an envvar does, so try this:

```
PYTHONTRACEMALLOC=100 pytest testing/test_warnings.py -k resource
```

Reply:
Ahh good catch. 😁 

Updated the test, thanks.